tests/Avalonia.Controls.UnitTests/TextBoxTests.cs

```diff
@@ -2129,6 +2129,25 @@ public void Losing_Focus_Should_Not_Reset_Selection()
             }
         }
 
+        [Fact]
+        public void Backspace_Should_Delete_Newline_Character_At_Once()
+        {
+            using var _ = UnitTestApplication.Start(Services);
+            var target = new TextBox
+            {
+                Template = CreateTemplate(),
+                Text = $"First{Environment.NewLine}Second",
```

Comment:
The test should probably not use platform specific line endings, as the code being tested does not. Have a test for \n, \r\n and \r?